code/datums/elements/food/food_trash.dm

```diff
@@ -38,17 +38,14 @@
 	INVOKE_ASYNC(src, .proc/async_generate_trash, source)
 
 /datum/element/food_trash/proc/async_generate_trash(datum/source)
-
-	var/obj/item/trash_item =  generate_trash_callback ? generate_trash_callback.Invoke(source) : new trash()
-
 	var/atom/edible_object = source
 
-	var/mob/living/mob_location = edible_object.loc //The foods location
+	var/obj/item/trash_item = generate_trash_procpath ? call(source, generate_trash_procpath)() : new trash(edible_object.drop_location())
 
-	if(istype(mob_location))
-		mob_location.put_in_hands(trash_item)
-	else
-		trash_item.forceMove(get_turf(edible_object))
+	if(istype(edible_object.loc))
```

Comment:
```suggestion
	if(isliving(edible_object.loc))
```

This line isn't correct, and will break if the edible object isn't on a living.

Reply:
![image](https://user-images.githubusercontent.com/24975989/110742563-8667fb00-822e-11eb-8259-7fffff06d8f9.png)
:eyes:

Reply:
That moment you don't push a change.

Reply:
and here i thought i had an original thought for a moment